cptv-reprocess.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python
+
+import argparse
+
+from api import API
+
+
+class CPTVReprocessor:
+    def __init__(self, server_url, user, password, id, limit):
+        self.user = user
+        self.password = password
+        self.url = server_url
+        self.recording_id = id
+        self.limit = limit
+
+    def reprocess(self):
+        """ Reprocesses all specified recording from specified server """
+
+        api = API(self.url, self.user, self.password)
+
+        print(f"Querying server {self.url}")
+        print(f"Limit is {self.limit}")
+        if self.recording_id:
+            print(f"Recording id is {self.recording_id}")
+            api.reprocess([self.recording_id])
+            return
+
+        where = {}
+        # algorithm was only added when trackbased tagging was introduced
+        where["additionalMetadata.algorithm"] = {"$eq": None}
+        where["type"] = "video"
+        count = api.query(where=where, limit=self.limit, raw_json=True)["count"]
+        if count:
+            print(f"Still reprocessing {count} records")
+            return
```

Comment:
I don't quite understand what the counting part is about. Aren't we expecting there to be recordings with additionalMetadata.algorithm not set?

Reply:
Oh ye its mean't to be where they aren't FINISHED